services/payments/src/simcore_service_payments/db/payments_transactions_repo.py

```diff
@@ -1,9 +1,190 @@
-    # services/web/server/src/simcore_service_webserver/payments/_db.py
-    # here.
-    # The transition should put all the implementations in the simcore_postgres_database first
-    # so it is usable temporarily in both the webserver and here
-    #
+import datetime
+from decimal import Decimal
+
+import sqlalchemy as sa
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from pydantic import HttpUrl, PositiveInt, parse_obj_as
+from pydantic.errors import PydanticErrorMixin
+from simcore_postgres_database.models.payments_transactions import (
+    PaymentTransactionState,
+    payments_transactions,
+)
+from sqlalchemy.ext.asyncio import AsyncEngine
+
+from ..models.db import PaymentsTransactionsDB
+from ..models.payments_gateway import PaymentID
+
+#
+# Errors
+#
+
+
+class PaymentsValueError(PydanticErrorMixin, ValueError):
+    msg_template = "Error in payment transaction '{payment_id}'"
+
+
+class PaymentNotFoundError(PaymentsValueError):
+    msg_template = "Payment transaction '{payment_id}' was not initialized."
+
+
+class PaymentAlreadyAckedError(PaymentsValueError):
     ...
+
+
+class PaymentAlreadyExistsError(PaymentsValueError):
+    msg_template = "Payment transaction '{payment_id}' was already initialized."
+
+
+#
+# base
+#
+
+
+class BaseRepository:
+    """
+    Repositories are pulled at every request
+    """
+
+    def __init__(self, db_engine: AsyncEngine):
+        assert db_engine is not None  # nosec
+        self.db_engine = db_engine
+
+
+#
+# repo
+#
+
+
+class PaymentsTransactionsRepo(BaseRepository):
+    async def insert_init_payment_transaction(
+        self,
+        payment_id: PaymentID,
+        price_dollars: Decimal,
+        osparc_credits: Decimal,
+        product_name: str,
+        user_id: UserID,
+        user_email: str,
+        wallet_id: WalletID,
+        comment: str | None,
+        initiated_at: datetime.datetime,
+    ) -> PaymentID:
+        """Annotates init-payment transaction"""
+        async with self.db_engine.begin() as conn:
+            await conn.execute(
+                payments_transactions.insert().values(
+                    payment_id=f"{payment_id}",
+                    price_dollars=price_dollars,
+                    osparc_credits=osparc_credits,
+                    product_name=product_name,
+                    user_id=user_id,
+                    user_email=user_email,
+                    wallet_id=wallet_id,
+                    comment=comment,
+                    initiated_at=initiated_at,
+                )
+            )
+            return payment_id
+
+    async def update_ack_payment_transaction(
+        self,
+        payment_id: PaymentID,
+        completion_state: PaymentTransactionState,
+        state_message: str | None,
+        invoice_url: HttpUrl,
+    ) -> PaymentsTransactionsDB:
+        """
+        - ACKs payment by updating state with SUCCESS, ...
```

Comment:
Well this is not true. You can pass any completition_state to this function.
If you want the comment to be valid remove the parameter and hardcode the desired state

Reply:
there are three dots after `...` :-)